design-documents/scalars-for-templates.md

```diff
@@ -143,41 +153,20 @@ interface FilterProcessorInterface
 ```
  
 _Template_ class would receive a list of such filters and use them accordingly.
-
-####### Alternatively:
-Use twig, it's an established and well known template engine and will solve this problems, we would just have
-to limit template variables to scalars.
+ 
+###### Deprecate old methods
+Old public methods following _\<Directive Name\>Directive()_ pattern must be deprecated in favor of new directive
+processors.
 
 #### Backward compatibility
 This would not be a backward compatible change - we use objects in our own
 E-mail templates in core Magento and developers are using objects in their own templates
 as well.
+
+In order to preserve backward compatibility E-mail templates read from _.html_ files will keep workings
```

Comment:
What will be the deprecation strategy? How do we stop supporting legacy style?

Reply:
added a bit about notifying developers about new restrictions and keeping old variables accessible for existing templates. Also the documents states in the end that support for legacy templates will be removed next minor version after these restrictions are introduced (2.5?)